# Copyright (C) 2019-2020 LEIDOS.

# Licensed under the Apache License, Version 2.0 (the "License"); you may not
# use this file except in compliance with the License. You may obtain a copy of
# the License at
 
# http://www.apache.org/licenses/LICENSE-2.0
 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
# License for the specific language governing permissions and limitations under
# the License.
 
# Double: The length of the trajectory in time domain, in seconds
# Units: Seconds
trajectory_time_length: 6.0

# String: The default control plugin name
control_plugin_name: "mpc_follower"


# Double : Minimum allowable speed
# Units : m/s
minimum_speed : 2.2352

# Double : Maximum allowable longitudinal acceleration 
# Units : m/s^2
max_accel : 1.5

# Double : Minimum value for lookahead distance 
# Units : meters
minimum_lookahead_distance : 5.0

# Double : Maximum value for lookahead distance 
# Units : meters
maximum_lookahead_distance : 25.0

# Double : Minimum speed value for lookahead calculation
# Units : m/s
minimum_lookahead_speed : 2.8

# Double : Maximum speed value for lookahead calculation
# Units : m/s
maximum_lookahead_speed : 13.9

# Double : Maximum allowable lateral acceleration 
# Units : m/s^2
lateral_accel_limit : 1.5

# Int : Size of the window used in the moving average filter to smooth both the computed curvature and output speeds
speed_moving_average_window_size : 5

#Int: // Size of the window used in the moving average filter to smooth the curvature profile
curvature_moving_average_window_size : 9

# Int : Number of points to look ahead when calculating the curvature of the lanelet centerline
curvature_calc_lookahead_count : 1

# Int: Amount to downsample input lanelet centerline data. Value corresponds to saving each nth point.
downsample_ratio : 8

# bool: Flag to activate and deactivate object avoidance communication with Yield
enable_object_avoidance_lc : false

# Double: The minimum timestep used for planning trajectory
#Units: seconds
min_timestep : 0.1

# Double : The additional downtrack beyond requested end dist used to fit points along spline. Extra points removed before returning trajectory. Feature not used if end dist is route end
#Units: meters
ending_buffer_downtrack : 5.0